tls/client-tls-smtp-overssl.c

```diff
@@ -0,0 +1,549 @@
+        fprintf(stderr, "ERROR: Failed to set the file descriptor\n");
+        goto cleanup;
+    }
+
+    /* Connect to wolfSSL on the server side */
+    if ((ret = wolfSSL_connect(ssl)) != SSL_SUCCESS) {
+        fprintf(stderr, "ERROR: failed to connect to wolfSSL\n");
+        goto cleanup;
+    }
+
+    /* Read the server data into our buff array */
+    memset(buff, 0, sizeof(buff));
+    if ((ret = wolfSSL_read(ssl, buff, sizeof(buff)-1)) == -1) {
+        fprintf(stderr, "ERROR: failed to read\n");
+        goto cleanup;
+    }
+    /* Compare if the response is right code or not */
+    if (!strncmp(buff, oversslCmd[0],strlen(oversslCmd[0]))) {
+        printf("%s\n", buff);
+    } else {
+        fprintf(stderr,"%s\n",buff);
+        goto cleanup;
+    }
+
+    /* Send "EHLO mail.example.com\r\n" to the server */
+    len =strlen(oversslCmd[1]);
+    if ((ret = wolfSSL_write(ssl, oversslCmd[1], len)) != len) {
+        fprintf(stderr, "ERROR: failed to send command.\n");
+        fprintf(stderr, "%d bytes of %d bytes were sent", ret, (int) len);
+        goto cleanup;
+    }
+
+    /* Read the server data into our buff array */
+    memset(buff, 0, sizeof(buff));
+    if ((ret = wolfSSL_read(ssl, buff, sizeof(buff)-1)) == -1) {
+        fprintf(stderr, "ERROR: failed to read\n");
+        goto cleanup;
+    }
+    /* Compare if the response is right code or not */
+    if (!strncmp(buff, oversslCmd[2],strlen(oversslCmd[2]))) {
+        printf("%s\n", buff);
+    } else {
+        fprintf(stderr,"ERROR: incorrect command received\n");
+        printf("%s\n", buff);
+        goto cleanup;
+    }
+
+    /* Send "AUTH LOGIN\r\n" to the server */
+    len = strlen(oversslCmd[3]);
+    if ((ret = wolfSSL_write(ssl, oversslCmd[3], len)) != len) {
+        fprintf(stderr, "ERROR: failed to send command.\n");
+        fprintf(stderr, "%d bytes of %d bytes were sent", ret, (int) len);
+        goto cleanup;
+    }
+
+    /* Read the server data into our buff array */
+    memset(buff, 0, sizeof(buff));
+    if ((ret = wolfSSL_read(ssl, buff, sizeof(buff)-1)) == -1) {
+        fprintf(stderr, "ERROR: failed to read\n");
+        goto cleanup;
+    }
+    /* Compare if the response is right code or not */
+    if (!strncmp(buff, oversslCmd[4],strlen(oversslCmd[4]))) {
+        printf("%s\n", buff);
+    } else {
+        fprintf(stderr,"ERROR: incorrect command received\n");
+        printf("%s\n", buff);
+        goto cleanup;
+    }
+
+    /* Get the mail address */
+    printf("Mail Address: ");
+    memset(plain,0,sizeof(plain));
+    if (fgets(plain, sizeof(plain), stdin) == NULL) {
+        fprintf(stderr, "ERROR: failed to get mail address.\n");
+        ret = -1;
+        goto cleanup;
+    }
+    /* Get the right mail address length */
+    for(len=0;len<sizeof(plain);len++){
+        if(plain[len] =='\n') break;
+    }
+
+    /* Encode the mail to Base64 */
+    word32 outLen = sizeof(buff);
```

Comment:
Can't mix variable declarations in code. Must be top of function or in a brace section.